test: add nic validation in import vm test

shwstppr opened the pull request:
### Description

Add smoke test to prevent any regression such as #6951.

Test now adds three networks to the VM, unmanages the VM and then imports the VM to verify that it returns all 3 NICs and other relevant VM details.

### Types of changes

- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Enhancement (improves an existing feature and functionality)
- [x] Cleanup (Code refactoring and cleanup, that may add test cases)

### Feature/Enhancement Scale or Bug Severity

#### Feature/Enhancement Scale

- [ ] Major
- [ ] Minor

#### Bug Severity

- [ ] BLOCKER
- [ ] Critical
- [ ] Major
- [ ] Minor
- [ ] Trivial


### Screenshots (if appropriate):


### How Has This Been Tested?
<!-- Please describe in detail how you tested your changes. -->
<!-- Include details of your testing environment, and the tests you ran to -->
<!-- see how your change affects other areas of the code, etc. -->


<!-- Please read the [CONTRIBUTING](https://github.com/apache/cloudstack/blob/main/CONTRIBUTING.md) document -->
```
[root@pr6953-t5437-vmware-67u3-marvin marvin]# nosetests --with-xunit --xunit-file=results.xml --with-marvin --marvin-config=./pr6953-t5437-vmware-67u3-advanced-cfg -s -a tags=advanced --hypervisor=VMware tests/smoke/test_vm_lifecycle_unmanage_import.py 
/usr/local/lib/python3.6/site-packages/paramiko/transport.py:33: CryptographyDeprecationWarning: Python 3.6 is no longer supported by the Python core team. Therefore, support for it is deprecated in cryptography and will be removed in a future release.
  from cryptography.hazmat.backends import default_backend

==== Marvin Init Started ====

=== Marvin Parse Config Successful ===

=== Marvin Setting TestData Successful===

==== Log Folder Path: /marvin/MarvinLogs/Dec_09_2022_08_44_09_MWLXVV All logs will be available here ====

=== Marvin Init Logging Successful===

==== Marvin Init Successful ====
=== TestName: test_01_unmanage_vm_cycle | Status : SUCCESS ===

=== Final results are now copied to: /marvin//MarvinLogs/test_vm_lifecycle_unmanage_import_PWF31S ===
```

rohityadavcloud commented:
@blueorangutan package

blueorangutan commented:
@rohityadavcloud a Jenkins job has been kicked to build packages. It will be bundled with  KVM, XenServer and VMware SystemVM templates. I'll keep you posted as I make progress.

blueorangutan commented:
Packaging result: :heavy_check_mark: el7 :heavy_check_mark: el8 :heavy_check_mark: debian :heavy_check_mark: suse15. SL-JID 4825

rohityadavcloud commented:
@blueorangutan test

blueorangutan commented:
@rohityadavcloud a Trillian-Jenkins test job (centos7 mgmt + kvm-centos7) has been kicked to run smoke tests

blueorangutan commented:
<b>Trillian test result (tid-5438)</b>
Environment: kvm-centos7 (x2), Advanced Networking with Mgmt server 7
Total time taken: 39179 seconds
Marvin logs: https://github.com/blueorangutan/acs-prs/releases/download/trillian/pr6964-t5438-kvm-centos7.zip
Smoke tests completed. 99 look OK, 2 have errors, 0 did not run
Only failed and skipped tests results shown below:


Test | Result | Time (s) | Test File
--- | --- | --- | ---
test_08_upgrade_kubernetes_ha_cluster | `Failure` | 574.64 | test_kubernetes_clusters.py
test_01_unmanage_vm_cycle | `Error` | 33.93 | test_vm_lifecycle_unmanage_import.py
ContextSuite context=TestUnmanageVM>:teardown | `Error` | 33.97 | test_vm_lifecycle_unmanage_import.py

shwstppr commented:
> Trillian test result (tid-5438) Environment: kvm-centos7 (x2), Advanced Networking with Mgmt server 7 Total time taken: 39179 seconds Marvin logs: https://github.com/blueorangutan/acs-prs/releases/download/trillian/pr6964-t5438-kvm-centos7.zip Smoke tests completed. 99 look OK, 2 have errors, 0 did not run Only failed and skipped tests results shown below:
> Test 	Result 	Time (s) 	Test File
> test_08_upgrade_kubernetes_ha_cluster 	`Failure` 	574.64 	test_kubernetes_clusters.py
> test_01_unmanage_vm_cycle 	`Error` 	33.93 	test_vm_lifecycle_unmanage_import.py
> ContextSuite context=TestUnmanageVM>:teardown 	`Error` 	33.97 	test_vm_lifecycle_unmanage_import.py

@rohityadavcloud I introduced an issue with test in this PR. I've fixed this in #6969
